pkg/nsx/services/mediator/mediator.go

```diff
@@ -82,7 +82,16 @@ func (serviceMediator *ServiceMediator) GetAvailableSubnet(subnetSet *v1alpha1.S
 		tags = append(tags, model.Tag{Scope: common.String(k), Tag: common.String(v)})
 	}
 	log.Info("the existing subnets are not available, creating new subnet", "subnetList", subnetList, "subnetSet.Name", subnetSet.Name, "subnetSet.Namespace", subnetSet.Namespace)
-	return serviceMediator.CreateOrUpdateSubnet(subnetSet, tags)
+	vpcList := serviceMediator.GetVPCsByNamespace(subnetSet.Namespace)
```

Comment:
this section seems the same as subnet_controller, is it possible to use a method instead of duplicate logic block

Reply:
done.